File: Calculate Money in Leetcode Bank/Calculate_Money_in_Leetcode_Bank.java
class Solution {
    public int totalMoney(int n) {
        int week = n / 7;
        int days = n % 7;
        int total = 0;

        total += 28 * week + 7 * (week * (week - 1) / 2);

        int startDay = week + 1;
        
        for (int i = 0; i < days; i++) {
            total += startDay + i;
        }
        return total;
    }
}

File: Calculate Money in Leetcode Bank/Calculate_Money_in_Leetcode_Bank.js
/**
 * @param {number} n
 * @return {number}
 */
var totalMoney = function (n) {
    var week = Math.floor(n / 7);
    var days = n % 7;
    var total = Number(0);

    total += 28 * week + 7 * (week * (week - 1) / 2);

    var startDay = week + 1;

    for (var i = 0; i < days; i++) {
        total += startDay + i;
    }

    return total;
};